src/bin/install.rs

```diff
@@ -0,0 +1,130 @@
+use std::path::Path;
+
+use cargo::ops;
+use cargo::core::{SourceId, GitReference};
+use cargo::util::{CliResult, Config, ToUrl, human};
+
+#[derive(RustcDecodable)]
+struct Options {
+    flag_jobs: Option<u32>,
+    flag_features: Vec<String>,
+    flag_no_default_features: bool,
+    flag_debug: bool,
+    flag_bin: Vec<String>,
+    flag_example: Vec<String>,
+    flag_verbose: bool,
+    flag_quiet: bool,
+    flag_color: Option<String>,
+    flag_root: Option<String>,
+    flag_list: bool,
+
+    arg_crate: Option<String>,
+    flag_vers: Option<String>,
+
+    flag_git: Option<String>,
+    flag_branch: Option<String>,
+    flag_tag: Option<String>,
+    flag_rev: Option<String>,
+
+    flag_path: Option<String>,
+}
+
+pub const USAGE: &'static str = "
+Install a Rust binary
+
+Usage:
+    cargo install [options] [<crate>]
+    cargo install [options] --list
+
+Specifying what crate to install:
+    --vers VERS               Specify a version to install from crates.io
+    --git URL                 Git URL to install the specified crate from
+    --branch BRANCH           Branch to use when installing from git
+    --tag TAG                 Tag to use when installing from git
+    --rev SHA                 Specific commit to use when installing from git
+    --path PATH               Filesystem path to local crate to install
+
+Build and install options:
+    -h, --help                Print this message
+    -j N, --jobs N            The number of jobs to run in parallel
+    --features FEATURES       Space-separated list of features to activate
+    --no-default-features     Do not build the `default` feature
+    --debug                   Build in debug mode instead of release mode
+    --bin NAME                Only install the binary NAME
+    --example EXAMPLE         Install the example EXAMPLE instead of binaries
+    --root DIR                Directory to install packages into
+    -v, --verbose             Use verbose output
+    -q, --quiet               Less output printed to stdout
+    --color WHEN              Coloring: auto, always, never
+
+This command manages Cargo's local set of install binary crates. Only packages
+which have [[bin]] targets can be installed, and all binaries are installed into
+the installation root's `bin` folder. The installation root is determined, in
+order of precedence, by `--root`, `$CARGO_INSTALL_ROOT`, the `install.root`
+configuration key, and finally the home directory (which is either
+`$CARGO_HOME` if set or `$HOME/.cargo` by default).
```

Comment:
I'd like to note my continuing discomfort with inventing Yet Another Language-Specific Homedir Install Location, rather than using ~/.local/, which is intended for user-scoped installation.